static/js/message_events.js

```diff
@@ -420,6 +424,10 @@ export function update_messages(events) {
         // produce correct results.
         compose_fade.update_message_list();
     }
+    // revive the lost focus on message edit due to rerender
+    if (message_under_update) {
```

Comment:
Is this only necessary in some subset of cases, like `changed_narrow` or `topic_edited` or similar?

Reply:
(Also, I suppose, does this belong in the `message_list_view.js` code path instead?)

Reply:
Yeah, probably you are right, this seemed to belong to `message_list_view` to me as well. But then I considered changing it keeping in mind that we want to fix this bug for message events, so I decided to put it back here. Thanks for the review, will make the necessary changes.

Reply:
@timabbott I think this is the solution of what we need, after trying out different things and using a debugger on every essential step, I realized while rendering line 404 here is removing the edit box and line 408 is re-rendering it, though it does not make much sense from the editor's perspective, we are essentially re-rendering the edit box and hence the focus is moving away, I don't think it makes sense to apply this focus logic sending parameters inside the render function because it is an edge specific to message update events. Please have a look and up for suggestions.

Reply:
Well, what if the algorithm is that `rerender_messages` checked if an element has focus when the rerender begins, and is destroyed/recreated by the rerender, then focus should be restarted?

@showell @ryanreh99 @amanagr thoughts on this algorithm question?

Reply:
I feel that can be a good approach, for all kinds of rerenders as a whole.

Reply:
Yeah, let's see how that looks like.